src/utils.js

```diff
@@ -99,34 +101,21 @@ export function defaultFieldValue(formData, schema) {
   return typeof formData === "undefined" ? schema.default : formData;
 }
 
-export function getAlternativeWidget(
-  schema,
-  widget,
-  registeredWidgets={},
-  widgetOptions={}
-) {
+export function getAlternativeWidget(schema, widget, registeredWidgets={}) {
   const {type, format} = schema;
 
-  function setDefaultOptions(widget) {
```

Comment:
Extending defaultProps was persistent across all instances of a widget, which was a bug. I fixed that by creating the wrapper element. Merging top level defaultProps is not needed there, since React by default merges those. Only options should be merged here. I documented that behaviour, because merging variables is not the React default behaviour.